x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/action/UpdateConnectorFilteringAction.java

```diff
@@ -70,8 +97,29 @@ public ActionRequestValidationException validate() {
                 validationException = addValidationError("[connector_id] cannot be [null] or [\"\"].", validationException);
             }
 
+            // If [filtering] is not present in the request payload it means that the user should define [rules] and/or [advanced_snippet]
             if (filtering == null) {
-                validationException = addValidationError("[filtering] cannot be [null].", validationException);
+                if (rules == null && advancedSnippet == null) {
+                    validationException = addValidationError("[advanced_snippet] and [rules] cannot be both [null].", validationException);
+                } else if (rules != null) {
+                    if (rules.isEmpty()) {
+                        validationException = addValidationError("[rules] cannot be an empty list.", validationException);
+                    } else if (isDefaultRulePresentInFilteringRules(rules) == false) {
```

Comment:
From a UX perspective: should we simply add the default rule as last rule, if it's missing? Or always add it and raise an error, if it's present in the request (indicating a change on the default rule). I could see it being cumbersome always adding the default rule by hand through the API, if you want to update basic rules. WDYT?

Reply:
That's a good suggestion, I will address this in a followup PR (should be easy to prepare on Monday). The missing logic from this PR is blocking me from opening PR related to draft validation and activation.